src/main/java/pl/piekoszek/gorskimatches/equation/EquationGenerator.java

```diff
@@ -27,5 +28,18 @@ public class EquationGenerator {
         }
         Set<String> keySet = allEquations.keySet();
         listOfMathCorrectEquations = new ArrayList<>(keySet);
+        for (String solution : listOfMathCorrectEquations) {
+            var mathCorrectEquationWithSolution = solutionToQuizzesMapper.insideSingleMatch(solution);
+            var quizFromMathCorrectEquation = mathCorrectEquationWithSolution.keySet();
+            for (String quiz : quizFromMathCorrectEquation) {
+                var solutionFromAllMathCorrectEquations = mathCorrectEquationWithSolution.get(quiz);
+                allQuizzesAndSolutions.putIfAbsent(quiz, new HashSet<>());
+                allQuizzesAndSolutions.get(quiz).addAll(solutionFromAllMathCorrectEquations);
```

Comment:
if changed to `allQuizzesAndSolutions.get(quiz).addAll(solutionsByEquation.get(quiz));` then `solutionFromAllMathCorrectEquations` variable could be removed